### PR TITLE
feat(datetime-button): add selecting interaction support

### PR DESCRIPTION
**Pull request by liamdebeasi**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue URL: Internal ticket

## What is the new behavior?

- Adds support for selecting either the date/time button and reflecting that selection by highlighting the button text
- Datetime button can now link with a datetime instance
- Datetime button now renders the localized date and time according to the `value` of datetime.
- Adds tests

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

There are a few things that will be addressed separately:

- The overlays are not correctly sized
- preferWheel is not fully supported with the combined date/time wheel picker
- Visual regression tests
- Disabling buttons, customizing buttons, etc

### CODE REVIEW

File: core/src/components/datetime-button/datetime-button.tsx [hunk 1]
```diff
@@ -40,8 +48,197 @@ export class DatetimeButton implements ComponentInterface {
...
+        case 'month-year':
+        case 'month':
+        case 'year':
+          this.selectedButton = 'date';
+          break;
+        case 'time-date':
+        case 'time':
+          this.selectedButton = 'time';
+          break;
+        default:
```
Comment (sean-perkins) [inline core/src/components/datetime-button/datetime-button.tsx:121]: ~ If there is no default case behavior, do we need to include it?

File: core/src/components/datetime-button/datetime-button.tsx [hunk 2]
```diff
@@ -40,8 +48,197 @@ export class DatetimeButton implements ComponentInterface {
...
+      case 'month-year':
+        this.dateText = getMonthAndYear(locale, parsedDatetime);
+        break;
+      case 'month':
+        this.dateText = getLocalizedDateTime(locale, parsedDatetime, { month: 'long' });
+        break;
+      case 'year':
+        this.dateText = getLocalizedDateTime(locale, parsedDatetime, { year: 'numeric' });
+        break;
+      default:
```
Comment (sean-perkins) [inline core/src/components/datetime-button/datetime-button.tsx:169]: ~ Same comment regarding default case.

**Review by sean-perkins — APPROVED** [review]

File: core/src/components/datetime-button/datetime-button.tsx (hunk 1 above)
Comment (liamdebeasi) [reply core/src/components/datetime-button/datetime-button.tsx:121]: Looks like it is optional: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch I will remove it

**Comment by amandaejohnston** [comment]
There seems to be something wonky with the time display; if I pick a different date, the displayed hour changes but the selected one in the wheel picker doesn't: https://user-images.githubusercontent.com/90629384/178825253-7d164e5c-8505-4b40-b439-9f0065248bad.mp4

File: core/src/components/datetime-button/datetime-button.tsx [hunk 3]
```diff
@@ -40,8 +48,189 @@ export class DatetimeButton implements ComponentInterface {
...
+    if (!datetime) {
+      printIonError(
+        'An ID associated with an ion-datetime instance is required for ion-datetime-button to function properly.'
+      );
+      return;
+    }
+
+    const datetimeEl = (this.datetimeEl = document.getElementById(datetime) as HTMLIonDatetimeElement | null);
+    if (!datetimeEl) {
+      printIonError(`No ion-datetime instance found for ID '${datetime}'.`);
```
Comment (amandaejohnston) [inline core/src/components/datetime-button/datetime-button.tsx:62]: ~ Could we include `this.el` in these error messages, for easier debugging?

**Pull request merged**